desktop/core/src/desktop/lib/rest/raz_http_client.py

```diff
@@ -47,6 +49,12 @@ def execute(self, http_method, path, params=None, data=None, headers=None, allow
     sig=pJL%2FWyed41tptiwBM5ymYre4qF8wzrO05tS5MCjkutc%3D&st=2015-01-02T01%3A40%3A51Z&se=2015-01-02T02%3A00%3A51Z&sp=r
     """
 
+    if '%' in path:
```

Comment:
Why this condition? If you need the path to be encoded in your url then why not always encode it?

Reply:
Good catch! I tested in live cluster and there's no harm in doing it for all path. I've done the necessary changes but I've found a small bug (not related to this) in which we are not able to go in the `%20` directories (although few other operations are working fine for them). So, I'm marking this PR as draft for now!

Reply:
Sounds good

Reply:
It looks that the condition is still present in the merged code. Did you forget or was there a good reason for adding it? In general, I strongly feel we should should avoid conditional encoding and stick with a consistent strategy: I.e. if the string is passed as a url parameter and **might** contain characters that need encoding, then we should always encode it. Conditionally encoding these strings will make future debugging much harder...